packages/react/src/file-input/FileInput.tsx

```diff
@@ -58,7 +58,7 @@ export const FileInput = forwardRef<HTMLInputElement, FileInputProps>(
 			buttonSize = 'sm',
 			label,
 			hideOptionalLabel,
-			required,
+			required: requiredProp,
```

Comment:
I think this `Prop` suffix convention we have should be removed. It breaks the recognition over recall heuristic and the self-documenting principle. When you're on line 161 it's easy to miss the line that massages the value back to what you want the prop to be which makes debugging a pain. Less so in this component but I've experienced it others with more complex massaging. I would prefer to see either:

1. Change the `Field` component so that it accepts `undefined` for `required` or,
2. Change this line to `required = false,` and remove line 141 or.

Reply:
I considered making `required = false` but then that doesn't guarante that people put a boolean still. Perhaps we should update Field to make it allow undefined

Reply:
Yeah, good point. I think updating Field is the way to go. It won't cause any breaking changes from my look through the code just now.